airflow/www/security.py

```diff
@@ -728,3 +735,102 @@ def check_authorization(
                 return False
 
         return True
+
+    # TODO: Whether to create APISecurityManager and move api related code to it?
+    def is_user_logged_in(self):
+        """Raise if user already logged in"""
+        if g.user is not None and g.user.is_authenticated:
+            raise Unauthenticated(detail="Client already authenticated")  # For security
+
+    def login_with_user_pass(self, username, password):
+        """Convenience method for user login through the API"""
+        self.is_user_logged_in()
+        if self.auth_type not in (AUTH_DB, AUTH_LDAP):
+            raise Unauthenticated(detail="Authentication type do not match")
+        user = None
+        if self.auth_type == AUTH_DB:
+            user = self.auth_user_db(username, password)
+        elif self.auth_type == AUTH_LDAP:
+            user = self.auth_user_ldap(username, password)
+        return user
+
+    def oauth_authorization_url(self, app, provider, redirect_url):
+        """Get authorization url for oauth"""
+        self.is_user_logged_in()
+        if self.auth_type != AUTH_OAUTH:
+            raise Unauthenticated(detail="Authentication type do not match")
+        state = jwt.encode(
+            request.args.to_dict(flat=False),
```

Comment:
Does this just include all request args from the user into the encode state output?

If so that's probably not a great idea and might be open to malicious activity in some way. What do we _need_ here?